guides/4)integrating_other_frameworks/Gradio_and_Wandb_Integration.md

```diff
@@ -0,0 +1,382 @@
+        assert os.path.exists(style_path), f"{style_path} does not exist!"
+
+
+        name = strip_path_extension(name)
+
+
+        # crop and align the face
+        style_aligned_path = os.path.join('style_images_aligned', f'{name}.png')
+        if not os.path.exists(style_aligned_path):
+            style_aligned = align_face(style_path)
+            style_aligned.save(style_aligned_path)
+        else:
+            style_aligned = Image.open(style_aligned_path).convert('RGB')
+
+
+        # GAN invert
+        style_code_path = os.path.join('inversion_codes', f'{name}.pt')
+        if not os.path.exists(style_code_path):
+            latent = e4e_projection(style_aligned, style_code_path, device)
+        else:
+            latent = torch.load(style_code_path)['latent']
+
+
+        targets.append(transform(style_aligned).to(device))
+        latents.append(latent.to(device))
+
+
+    targets = torch.stack(targets, 0)
+    latents = torch.stack(latents, 0)
+
+
+    target_im = utils.make_grid(targets, normalize=True, range=(-1, 1))
+    display_image(target_im, title='Style References')
+    ```
+
+
+5. Finetune StyleGAN and W&B experiment tracking
+
+    This next step will open a W&B dashboard to track your experiments and a gradio panel showing pretrained models to choose from a drop down menu from a Gradio Demo hosted on Huggingface Spaces.
+
+    ```python
+    #@title Finetune StyleGAN
+    #@markdown alpha controls the strength of the style
+    alpha =  1.0 #@param {type:"slider", min:0, max:1, step:0.1}
+    alpha = 1-alpha
+
+
+    #@markdown Tries to preserve color of original image by limiting family of allowable transformations. Set to false if you want to transfer color from reference image. This also leads to heavier stylization
+    preserve_color = True #@param{type:"boolean"}
+    #@markdown Number of finetuning steps. Different style reference may require different iterations. Try 200~500 iterations.
+    num_iter = 200 #@param {type:"number"}
+    #@markdown Log training on wandb and interval for image logging
+    use_wandb = True #@param {type:"boolean"}
+    log_interval = 50 #@param {type:"number"}
+
+
+    samples = []
+    column_names = ["Referece (y)", "Style Code(w)", "Real Face Image(x)"]
+
+
+    if use_wandb:
+        wandb.init(project="JoJoGAN")
+        config = wandb.config
+        config.num_iter = num_iter
+        config.preserve_color = preserve_color
+        wandb.log(
+        {"Style reference": [wandb.Image(transforms.ToPILImage()(target_im))]},
+        step=0)
+        wandb.log({"Gradio panel": wandb.Html('''
+    <link rel="stylesheet" href="https://gradio.s3-us-west-2.amazonaws.com/2.6.2/static/bundle.css">
+    <div id="JoJoGAN-demo"></div>
+    <script src="https://gradio.s3-us-west-2.amazonaws.com/2.6.2/static/bundle.js"></script>
+    <script>
+    launchGradioFromSpaces("akhaliq/JoJoGAN", "#JoJoGAN-demo")
+    </script>
+    <style>
+    /* work around a weird bug */
```

Comment:
Can we remove this comment

Reply:
removed comment and updated section for using web components